attributes/attributes.go

```diff
@@ -121,13 +122,78 @@ func (a *Attributes) String() string {
 	return sb.String()
 }
 
-func str(x any) string {
-	if v, ok := x.(fmt.Stringer); ok {
+const nilAngleString = "<nil>"
+
+func str(x any) (s string) {
+	defer func() {
+		if r := recover(); r != nil {
+			// If it panics with a nil pointer, just say "<nil>". The likeliest causes are a
+			// [fmt.Stringer] that fails to guard against nil or a nil pointer for a
+			// value receiver, and in either case, "<nil>" is a nice result.
+			//
+			// Adapted from the code in fmt/print.go.
+			if v := reflect.ValueOf(x); v.Kind() == reflect.Pointer && v.IsNil() {
+				s = nilAngleString
+				return
+			}
+
+			// The panic was likely not caused by fmt.Stringer.
+			panic(r)
+		}
+	}()
+	if x == nil { // NOTE: typed nils will not be caught by this check
+		return nilAngleString
+	} else if v, ok := x.(fmt.Stringer); ok {
 		return v.String()
```

Comment:
What if we don't call `v.String()` here, but instead call `fmt.Sprint(v)` or `fmt.Sprintf("%s", v)` or something similar. Would we be able to eliminate the handling of `panic` with a `recover`?

Reply:
+ Agree. Avoid duplicate the recover code from `fmt`, let Go handle the panic itself.
+ I think [panic](https://github.com/golang/go/blob/1fde99cd6eff725f5cc13748a43b4aef3de557c8/src/runtime/panic.go#L715)&[recover](https://github.com/golang/go/blob/1fde99cd6eff725f5cc13748a43b4aef3de557c8/src/runtime/panic.go#L979) is fast as the code is simple enough and `panic&recover` is on unlikely path. In fact, that's how `json.Marshal` works: [encode.go#L287](https://github.com/golang/go/blob/master/src/encoding/json/encode.go#L287)
```golang
// jsonError is an error wrapper type for internal use only.
// Panics with errors are wrapped in jsonError so that the top-level recover
// can distinguish intentional panics from this package.
type jsonError struct{ error }

func (e *encodeState) marshal(v any, opts encOpts) (err error) {
	defer func() {
		if r := recover(); r != nil {
			if je, ok := r.(jsonError); ok {
				err = je.error
			} else {
				panic(r)
			}
		}
	}()
	e.reflectValue(reflect.ValueOf(v), opts)
	return nil
}

// error aborts the encoding by panicking with err wrapped in jsonError.
func (e *encodeState) error(err error) {
	panic(jsonError{err})
}
```

Reply:
We are not too worried about performance with panic and recover. We don't want to have that complexity in our code when it is not absolutely required.

Reply:
patch added. 
+ use `fmt.Sprint(v)` instead of `v.String()` since it's more efficient than `fmt.Sprintf("%s", v)`.
+ `panic&recover `codes removed, `fmt` will take this case.